src/packages/documents/documents/modals/culture-and-hostnames/culture-and-hostnames-modal.element.ts

```diff
@@ -0,0 +1,235 @@
+import { UmbDocumentRepository } from '../../repository/document.repository.js';
+import { html, customElement, state, css, repeat, query } from '@umbraco-cms/backoffice/external/lit';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import { UmbModalBaseElement } from '@umbraco-cms/backoffice/modal';
+import { UUIInputEvent, UUIPopoverContainerElement, UUISelectEvent } from '@umbraco-cms/backoffice/external/uui';
+import { UmbLanguageRepository } from '@umbraco-cms/backoffice/language';
+import { DomainPresentationModel, LanguageResponseModel } from '@umbraco-cms/backoffice/backend-api';
+import { UmbCultureAndHostnamesModalData, UmbCultureAndHostnamesModalValue } from '@umbraco-cms/backoffice/document';
+
+@customElement('umb-culture-and-hostnames-modal')
+export class UmbCultureAndHostnamesModalElement extends UmbModalBaseElement<
+	UmbCultureAndHostnamesModalData,
+	UmbCultureAndHostnamesModalValue
+> {
+	#documentRepository = new UmbDocumentRepository(this);
+	#languageRepository = new UmbLanguageRepository(this);
+
+	#unique?: string | null;
+	#languageModel: Array<LanguageResponseModel> = [];
+
+	@state()
+	private _options: Array<Option> = [];
+
+	@state()
+	private _domains: Array<DomainPresentationModel> = [];
+
+	@query('#more-options')
+	popoverContainerElement?: UUIPopoverContainerElement;
+
+	#handleCancel() {
+		this.modalContext?.reject();
+	}
+	#handleSave() {
+		// TODO validation before submitting
+		this.#documentRepository.saveCultureAndHostnames(this.#unique!, this.value);
+		this.modalContext?.submit();
+	}
+
+	firstUpdated() {
+		this.#unique = this.data?.unique;
+		this.#getDomains();
+	}
+
+	async #getDomains() {
+		if (!this.#unique) return;
+		const { data } = await this.#documentRepository.getCultureAndHostnames(this.#unique);
+
+		if (!data) return;
+		this._domains = data.domains.map((domain) => ({ isoCode: domain.isoCode, domainName: domain.domainName }));
```

Comment:
Please make a repository specifically to work with document culture and host names. We need to split our repositories into domains to make reusability easier and to prevent extremely large repositories. You can add this repository into the culture and hostnames folder to keep the related functionality close to each other. The name of the repository could be somehing like `UmbDocumentCultureAndHostnamesRepository`.